cmd/embedded-cluster/join.go

```diff
@@ -185,36 +197,75 @@ var joinCommand = &cli.Command{
 			}
 		}
 
-		logrus.Infof("Joining node to cluster")
+		logrus.Debugf("joining node to cluster")
 		if err := runK0sInstallCommand(jcmd.K0sJoinCommand); err != nil {
 			err := fmt.Errorf("unable to join node to cluster: %w", err)
 			metrics.ReportJoinFailed(c.Context, jcmd.MetricsBaseURL, jcmd.ClusterID, err)
 			return err
 		}
 
-		logrus.Infof("Applying configuration overrides")
+		logrus.Debugf("applying configuration overrides")
 		if err := applyJoinConfigurationOverrides(jcmd); err != nil {
 			err := fmt.Errorf("unable to apply configuration overrides: %w", err)
 			metrics.ReportJoinFailed(c.Context, jcmd.MetricsBaseURL, jcmd.ClusterID, err)
 			return err
 		}
 
-		logrus.Infof("Creating systemd unit files")
+		logrus.Debugf("creating systemd unit files")
 		if err := createSystemdUnitFiles(jcmd.K0sJoinCommand); err != nil {
 			err := fmt.Errorf("unable to create systemd unit files: %w", err)
 			metrics.ReportJoinFailed(c.Context, jcmd.MetricsBaseURL, jcmd.ClusterID, err)
 			return err
 		}
 
-		logrus.Infof("Starting %s service", binName)
+		logrus.Debugf("starting %s service", binName)
 		if err := startK0sService(); err != nil {
 			err := fmt.Errorf("unable to start service: %w", err)
 			metrics.ReportJoinFailed(c.Context, jcmd.MetricsBaseURL, jcmd.ClusterID, err)
 			return err
 		}
 
+		if err := waitForK0s(); err != nil {
+			err := fmt.Errorf("unable to wait for node: %w", err)
+			metrics.ReportJoinFailed(c.Context, jcmd.MetricsBaseURL, jcmd.ClusterID, err)
+			return err
+		}
+
+		if !strings.Contains(jcmd.K0sJoinCommand, "controller") {
```

Comment:
does this mean we're not waiting for workers to join, only controllers?

Reply:
Yes, for right now. I had run into an issue with workers where I could not get a k8s client instantiated to do that check. Maybe we need to be looking elsewhere for the kubeconfig for workers?